src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -104,7 +104,7 @@ public virtual Type[] GetGenericParameterConstraints()
         public bool IsContextful => IsContextfulImpl();
         protected virtual bool IsContextfulImpl() => false;
 
-        public virtual bool IsEnum => IsSubclassOf(typeof(Enum));
+        public virtual bool IsEnum => IsValueType && IsAssignableTo(typeof(Enum));
```

Comment:
I was not sure whether it makes more sense to intrinsify IsEnum or to switch it to two intrinsic calls like this.

Reply:
I do not think you want to be changing the implementation of the abstract System.Type base class. It is a potential breaking change.

Reply:
Should I overload it in RuntimeType or make it an intrinsic then?

Reply:
Looks like on WASM IsAssignableTo ends up calling into IsEnum, think making it an intrinsic will be a better solution then.